test.sh

```diff
@@ -45,6 +45,17 @@ update_status() {
   fi
 }
 
+function error_on_output() {
```

Comment:
I probably wasn't clear enough here. Every place we currently call run_and_comment has the semantics that if it produces output, it's a failure. So it's unnecessary to add a special error_on_output function. Instead, just add these four lines in run_and_comment's `else` clause:

```
# Any output indicates a failure
FAILURE=1
update_status --state failure
echo "[!] FAILURE: $@"
```